flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/reader/SourceReaderBase.java

```diff
@@ -81,6 +82,8 @@ Licensed to the Apache Software Foundation (ASF) under one
 	/** The last element to ensure it is fully handled. */
 	private SplitsRecordIterator<E> splitIter;
 
+	private volatile boolean noMoreSplitsAssignment;
```

Comment:
Does this need to be volatile? Isn't it updated always by the same thread (mailbox thread)?